pyveda/frameworks/batch_generator.py

```diff
@@ -0,0 +1,112 @@
+import numpy as np
+
+from random import randint
+from random import sample
+
+from pyveda.frameworks.transforms import *
+
+class BaseGenerator():
+    '''
+    Parent Class for Generator
+
+    cache: Parition (train, test, or validate) of VedaBase or VedaStream.
+    batch_size: int. Number of samples.
+    shuffle: Boolean. To shuffle or not shuffle data between epochs.
+    Rescale: Boolean. Flag to indicate if data returned from the generator should be rescaled between 0 and 1.
+    '''
+
+    def __init__(self, cache, batch_size=32, shuffle=True, rescale=False):
+        self.cache = cache
+        self.batch_size = batch_size
+        self.index = 0
+        self.shuffle = shuffle
+        self.rescale = rescale
+        self.on_epoch_end()
+        self.list_ids = np.arange(0, len(self.cache))
+
+    @property
+    def mltype(self):
+        return self.cache._vset.mltype
+
+    @property
+    def shape(self):
+        return self.cache._vset.image_shape
+
+    def build_batch(self, index):
+        raise NotImplemented
+
+    def _empty_batch(self, batch_size):
+        x = np.empty((self.batch_size, *self.shape))
+        if self.mltype == 'classification':
+            y = np.empty((self.batch_size), dtype=int)   # needs classes
+        if self.mltype == 'segmentation':
+            y = np.empty((self.batch_size, *self.shape[1:]))
```

Comment:
in order to use shape here we need to define shape somewhere, im guessing from self.cache.image_shape

Reply:
shape got defined as a property on the `BaseGenerator` class does that not work?

Reply:
oh im sorry, i missed that, doh